3d_armor/api.lua

```diff
@@ -258,21 +270,42 @@ armor.set_player_armor = function(self, player)
 				local value = def.groups["armor_"..attr] or 0
 				attributes[attr] = attributes[attr] + value
 			end
-			local mat = string.match(item, "%:.+_(.+)$")
-			if material.name then
-				if material.name == mat then
-					material.count = material.count + 1
+		end
+	end
+	-- The following code compares player worn armor items against requirements
+	-- of which armor pieces are needed to be worn to meet set bonus requirements
+	for loc,item in pairs(worn_armor) do
+		local item_mat = string.match(item, "%:.+_(.+)$")
+		for k,set_loc in pairs(armor.config.set_elements)do
+			if set_loc == loc then
+				if item_mat ~= nil then
+					if set_worn[item_mat] == nil then
+						set_worn[item_mat] = 0
+						set_worn[item_mat] = set_worn[item_mat] + 1
+					else
+						set_worn[item_mat] = set_worn[item_mat] + 1
+					end
+				else
+					if set_worn["unknown"] == nil then
+						set_worn["unknown"] = 0
+						set_worn["unknown"] = set_worn["unknown"] + 1
+					else
+						set_worn["unknown"] = set_worn["unknown"] + 1
+					end
 				end
-			else
-				material.name = mat
 			end
 		end
 	end
+	for mat_name,arm_piece_num in pairs(set_worn) do
```

Comment:
Would you please be so nice to add comments for each step to clarify what's done?

Reply:
will do

Reply:
I just added this Smalljoker, I wasnt sure how much to comment as 3d_armor dosent have alot of commentary but Im more than happy to add more if thats accepatable. I've found unwravlling some segements of code to be quiet challanging.

```
	-- The following code checks if the player should recieve the set bonus/multiplyer
	-- by testing the table length of armor.config.set_elements (numeric key) and 
	-- player armor pieces worn (arm_piece_num) against each other if numbers 
	-- are equal the player is wearing a full set of armor. We don't need to 
	-- check locations here as that was performed at api.lua line 282 "set_loc == loc".
	for mat_name,arm_piece_num in pairs(set_worn) do
		if arm_piece_num == #armor.config.set_elements then
			armor_multi = armor.config.set_multiplier
		end
	end
```

Reply:
Commenting is always a good thing to do, as long it helps clarifying what happens.
I think this here would work as well, and is a bit shorter:
```Lua
-- Apply the armor multiplier only if the player is wearing a full set of armor
```
and to clarify what's happening near L282:
```Lua
-- Perform location checks to ensure the armor is worn correctly
for k,set_loc in pairs(armor.config.set_elements) do
```

Reply:
Updated as above :)